streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java

```diff
@@ -187,7 +185,8 @@ public void testReprocessingFromScratchAfterResetWithIntermediateUserTopic() thr
         streams.close();
 
         assertThat(resultRerun, equalTo(result));
-        assertThat(resultRerun2, equalTo(result2));
+        final int maxToCompare = Math.min(result2.size(), resultRerun2.size());
+        assertThat(resultRerun2.subList(0, maxToCompare), equalTo(result2.subList(0, maxToCompare)));
```

Comment:
@mjsax With the recent changes on IntegrationTestUtils, `waitUntilXXX` will use one consumer instance only so I'm wondering if it could still return more than expected results.

Also for the mock time issue, could we eliminate the non-determinism by not using `System.currentTimeMillis`? We can augment the `MockTime` to set manual initialization values if necessary.